src/STM32RTC.cpp

```diff
@@ -0,0 +1,814 @@
+    if(year < 100) {
+      _year = year;
+    }
+    RTC_SetDate(_year, _month, _date, _day);
+  }
+}
+
+/**
+  * @brief  set RTC calendar.
+  * @param  day: 1-31
+  * @param  month: 1-12
+  * @param  year: 0-99
+  * @retval none
+  */
+void STM32RTC::setDate(uint8_t day, uint8_t month, uint8_t year)
+{
+  if (_configured) {
+    setDay(day);
+    setMonth(month);
+    setYear(year);
+  }
+}
+
+/**
+  * @brief  set RTC calendar.
+  * @param  weekDay: 1-7 (Monday first)
+  * @param  day: 1-31
+  * @param  month: 1-12
+  * @param  year: 0-99
+  * @retval none
+  */
+void STM32RTC::setDate(uint8_t weekDay, uint8_t day, uint8_t month, uint8_t year)
+{
+  if (_configured) {
+    setWeekDay(weekDay);
+    setDay(day);
+    setMonth(month);
+    setYear(year);
+  }
+}
+
+/**
+  * @brief  set RTC alarm second.
+  * @param  seconds: 0-59
+  * @retval none
+  */
+void STM32RTC::setAlarmSeconds(uint8_t seconds)
+{
+  if (_configured) {
+    if(seconds < 60) {
+      _alarmSeconds = seconds;
+    }
+  }
+}
+
+/**
+  * @brief  set RTC alarm minute.
+  * @param  minutes: 0-59
+  * @retval none
+  */
+void STM32RTC::setAlarmMinutes(uint8_t minutes)
+{
+  if (_configured) {
+    if(minutes < 60) {
+      _alarmMinutes = minutes;
+    }
+  }
+}
+
+/**
+  * @brief  set RTC alarm hour.
+  * @param  hour: 0-23
+  * @retval none
+  */
+void STM32RTC::setAlarmHours(uint8_t hours)
+{
+  if (_configured) {
+    if(hours < 24) {
+      _alarmHours = hours;
+    }
+  }
+}
+
+/**
+  * @brief  set RTC alarm hour.
+  * @param  hour: 0-23 or 0-12
+  * @param  hour format: AM or PM
+  * @retval none
+  */
+void STM32RTC::setAlarmHours(uint8_t hours, hourAM_PM_t format)
+{
+  if (_configured) {
+    if(hours < 24) {
+      _alarmHours = hours;
+    }
+    _alarmFormat = format;
+  }
+}
+
+/**
+  * @brief  set RTC alarm time.
+  * @param  hours: 0-23
+  * @param  minutes: 0-59
+  * @param  seconds: 0-59
+  * @retval none
+  */
+void STM32RTC::setAlarmTime(uint8_t hours, uint8_t minutes, uint8_t seconds)
+{
+  if (_configured) {
+    setAlarmHours(hours);
+    setAlarmMinutes(minutes);
+    setAlarmSeconds(seconds);
+  }
+}
+
+/**
+  * @brief  set RTC alarm time.
+  * @param  hours: 0-23
+  * @param  minutes: 0-59
+  * @param  seconds: 0-59
+  * @param  hour format: AM or PM
+  * @retval none
+  */
+void STM32RTC::setAlarmTime(uint8_t hours, uint8_t minutes, uint8_t seconds, hourAM_PM_t format)
+{
+  if (_configured) {
+    setAlarmHours(hours, format);
+    setAlarmMinutes(minutes);
+    setAlarmSeconds(seconds);
+  }
+}
+
+/**
+  * @brief  set RTC alarm date.
+  * @param  day: 1-31
+  * @retval none
+  */
+void STM32RTC::setAlarmDay(uint8_t day)
+{
+  if (_configured) {
+    if((day >= 1) && (day <= 31)) {
+      _alarmDate = day;
+    }
+  }
+}
+
+// Kept for compatibility.
+void STM32RTC::setAlarmMonth(uint8_t month)
+{
+  UNUSED(month);
+}
+
+// Kept for compatibility.
+void STM32RTC::setAlarmYear(uint8_t year)
+{
+  UNUSED(year);
```

Comment:
please explain and document the implication of not using / supporting the year

Reply:
Done!